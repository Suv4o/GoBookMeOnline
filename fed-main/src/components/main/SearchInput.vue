<script lang="ts">
export default {
  name: 'SearchInput',
}
</script>

<script setup lang="ts">
import { computed, ref } from 'vue'
import { SearchIcon } from '@heroicons/vue/solid'
import { ChevronRightIcon, UsersIcon } from '@heroicons/vue/outline'
import { Combobox, ComboboxInput, ComboboxOptions, ComboboxOption } from '@headlessui/vue'

const people = [
  {
    id: 1,
    name: 'Leslie Alexander',
    phone: '1-493-747-9031',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 2,
    name: 'Juniper Moody',
    phone: '1-582-300-8610',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 3,
    name: 'Miyah Brady',
    phone: '1-316-938-5949',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 4,
    name: 'Ellen Walker',
    phone: '1-203-368-8795',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
  {
    id: 5,
    name: 'Emma Nicholson',
    phone: '1-248-393-7011',
    email: 'lesliealexander@example.com',
    role: 'Co-Founder / CEO',
    url: 'https://example.com',
    profileUrl: '#',
    imageUrl:
      'https://images.unsplash.com/photo-1494790108377-be9c29b29330?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80',
  },
]

// const recent = [people[0], people[1], people[2], people[3], people[4]]
const recent: any = []
const isFocused = ref(false)

const query = ref('')
const filteredPeople = computed(() =>
  query.value === ''
    ? []
    : people.filter(person => {
        return person.name.toLowerCase().includes(query.value.toLowerCase())
      })
)

function onSelect(person: any) {
  window.location = person.url
}

function setRecentSearch() {
  isFocused.value = true
}

function unsetRecentSearch() {
  isFocused.value = false
}
</script>

<template>
  <Combobox
    v-slot="{ activeOption }"
    as="div"
    class="transform divide-y divide-gray-100 overflow-hidden rounded-xl bg-white shadow-2xl ring-1 ring-black ring-opacity-5 transition-all"
    @update:model-value="onSelect"
  >
    <div class="relative">
      <SearchIcon class="pointer-events-none absolute top-3.5 left-4 h-5 w-5 text-gray-400" aria-hidden="true" />
      <ComboboxInput
        class="h-12 w-full border-0 bg-transparent pl-11 pr-4 text-gray-800 placeholder-gray-400 focus:ring-0 sm:text-sm"
        placeholder="Search..."
        @focus="setRecentSearch"
        @blur="unsetRecentSearch"
        @change="query = $event.target.value"
      />
    </div>

    <ComboboxOptions
      v-if="query === '' || filteredPeople.length > 0"
      class="flex divide-x divide-gray-100"
      as="div"
      static
      hold
    >
      <div
        v-if="isFocused && (query !== '' || recent.length)"
        :class="['max-h-96 min-w-0 flex-auto scroll-py-4 overflow-y-auto px-6 py-4', activeOption && 'sm:h-96']"
      >
        <h2 v-if="query === '' && recent.length" class="mt-2 mb-4 text-xs font-semibold text-gray-500">
          Recent searches
        </h2>
        <div hold class="-mx-2 text-sm text-gray-700">
          <ComboboxOption
            v-for="person in query === '' ? recent : filteredPeople"
            :key="person.id"
            v-slot="{ active }"
            :value="person"
            as="template"
          >
            <div
              :class="[
                'group flex cursor-default select-none items-center rounded-md p-2',
                active && 'bg-gray-100 text-gray-900',
              ]"
            >
              <img :src="person.imageUrl" alt="" class="h-6 w-6 flex-none rounded-full" />
              <span class="ml-3 flex-auto truncate">{{ person.name }}</span>
              <ChevronRightIcon v-if="active" class="ml-3 h-5 w-5 flex-none text-gray-400" aria-hidden="true" />
            </div>
          </ComboboxOption>
        </div>
      </div>

      <div
        v-if="activeOption"
        class="hidden h-96 w-1/2 flex-none flex-col divide-y divide-gray-100 overflow-y-auto sm:flex"
      >
        <div class="flex-none p-6 text-center">
          <img :src="activeOption.imageUrl" alt="" class="mx-auto h-16 w-16 rounded-full" />
          <h2 class="mt-3 font-semibold text-gray-900">
            {{ activeOption.name }}
          </h2>
          <p class="text-sm leading-6 text-gray-500">{{ activeOption.role }}</p>
        </div>
        <div class="flex flex-auto flex-col justify-between p-6">
          <dl class="grid grid-cols-1 gap-x-6 gap-y-3 text-sm text-gray-700">
            <dt class="col-end-1 font-semibold text-gray-900">Phone</dt>
            <dd>{{ activeOption.phone }}</dd>
            <dt class="col-end-1 font-semibold text-gray-900">URL</dt>
            <dd class="truncate">
              <a :href="activeOption.url" class="text-teal-600 underline">
                {{ activeOption.url }}
              </a>
            </dd>
            <dt class="col-end-1 font-semibold text-gray-900">Email</dt>
            <dd class="truncate">
              <a :href="`mailto:${activeOption.email}`" class="text-teal-600 underline">
                {{ activeOption.email }}
              </a>
            </dd>
          </dl>
          <button
            type="button"
            class="mt-6 w-full rounded-md border border-transparent bg-teal-600 py-2 px-4 text-sm font-medium text-white shadow-sm hover:bg-teal-700 focus:outline-none focus:ring-2 focus:ring-teal-500 focus:ring-offset-2"
          >
            Send message
          </button>
        </div>
      </div>
    </ComboboxOptions>

    <div v-if="query !== '' && filteredPeople.length === 0" class="py-14 px-6 text-center text-sm sm:px-14">
      <UsersIcon class="mx-auto h-6 w-6 text-gray-400" aria-hidden="true" />
      <p class="mt-4 font-semibold text-gray-900">No people found</p>
      <p class="mt-2 text-gray-500">We couldnâ€™t find anything with that term. Please try again.</p>
    </div>
  </Combobox>
</template>
