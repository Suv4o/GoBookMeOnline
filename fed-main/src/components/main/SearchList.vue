<script lang="ts">
export default {
  name: 'SearchList',
}
</script>

<script setup lang="ts">
import BookSearchList from '../global/BookSearchList/BookSearchList.vue'
import BookPagination from '../global/BookPagination/BookPagination.vue'
</script>

<template>
  <book-search-list title="Locations" type="location" />
  <book-search-list title="People" type="person" />
  <div class="mt-4">
    <book-pagination />
  </div>
</template>
