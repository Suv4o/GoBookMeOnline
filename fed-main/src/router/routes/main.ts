import type { RouteRecordRaw } from 'vue-router'

const mainRoutes: Array<RouteRecordRaw> = [
  {
    path: '/',
    name: 'home',
    components: {
      default: () => import('../../views/HomePage.vue'),
      DefaultMainNav: () => import('../../components/Default/DefaultMainNav/DefaultMainNav.vue'),
      DefaultNewsletter: () => import('../../components/Default/DefaultNewsletter/DefaultNewsletter.vue'),
      DefaultFooter: () => import('../../components/Default/DefaultFooter/DefaultFooter.vue'),
    },
  },
  {
    path: '/:pathMatch(.*)*',
    name: 'notFound',
    components: {
      default: () => import('../../views/NotFoundPage.vue'),
    },
  },
]

export default mainRoutes
