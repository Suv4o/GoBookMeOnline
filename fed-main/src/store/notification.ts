import { defineStore } from 'pinia'

export enum NotificationTypes {
  Success = 'success',
  Error = 'error',
  Warning = 'warning',
  Info = 'info',
}

export interface NotificationState {
  isOpen: boolean
  type: NotificationTypes | null
  title: string | null
  message: string | null
}

export const useNotificationStore = defineStore('notificationStore', {
  state: (): NotificationState => {
    return {
      isOpen: false,
      type: null,
      title: null,
      message: null,
    }
  },
})
