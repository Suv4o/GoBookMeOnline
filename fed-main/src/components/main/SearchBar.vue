<script lang="ts">
export default {
  name: 'SearchBar',
}
</script>

<script setup lang="ts">
import SearchInput from './SearchInput.vue'
import LocationInput from './LocationInput.vue'
</script>

<template>
  <div class="flex flex-row mx-auto max-w-3xl">
    <div class="basis-2/3 mr-4">
      <search-input />
    </div>
    <div class="basis-1/3"><location-input /></div>
  </div>
</template>
