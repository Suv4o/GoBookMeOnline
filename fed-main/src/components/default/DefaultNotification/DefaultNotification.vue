<script lang="ts">
export default {
  name: 'DefaultNotification',
}
</script>

<script setup lang="ts">
import { NotificationTypes } from '../../../store/notification'
import { CheckCircleIcon, XCircleIcon, ExclamationTriangleIcon, InformationCircleIcon } from '@heroicons/vue/24/outline'
import { XMarkIcon } from '@heroicons/vue/24/solid'
import { useNotificationStore } from '../../../store/notification'

const notificationStore = useNotificationStore()

function close() {
  notificationStore.isOpen = false
}
</script>

<template>
  <div
    aria-live="assertive"
    class="fixed inset-0 flex items-end px-4 py-6 pointer-events-none sm:p-6 sm:items-start z-30"
  >
    <div class="w-full flex flex-col items-center space-y-4 sm:items-end">
      <transition
        enter-active-class="transform ease-out duration-300 transition"
        enter-from-class="translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2"
        enter-to-class="translate-y-0 opacity-100 sm:translate-x-0"
        leave-active-class="transition ease-in duration-100"
        leave-from-class="opacity-100"
        leave-to-class="opacity-0"
      >
        <div
          v-if="notificationStore.isOpen"
          data-testid="Default Notification"
          class="max-w-sm w-full bg-white shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"
        >
          <div class="p-4">
            <div class="flex items-start">
              <div data-testid="Notification icon" class="flex-shrink-0">
                <CheckCircleIcon
                  v-if="notificationStore.type === NotificationTypes.Success"
                  class="h-6 w-6 text-green-400"
                  aria-hidden="true"
                />
                <XCircleIcon
                  v-if="notificationStore.type === NotificationTypes.Error"
                  class="h-6 w-6 text-red-400"
                  aria-hidden="true"
                />
                <ExclamationTriangleIcon
                  v-if="notificationStore.type === NotificationTypes.Warning"
                  class="h-6 w-6 text-yellow-400"
                  aria-hidden="true"
                />
                <InformationCircleIcon
                  v-if="notificationStore.type === NotificationTypes.Info"
                  class="h-6 w-6 text-blue-400"
                  aria-hidden="true"
                />
              </div>
              <div class="ml-3 w-0 flex-1 pt-0.5">
                <p data-testid="Notification title" class="text-sm font-medium text-gray-900">
                  {{ notificationStore.title }}
                </p>
                <p data-testid="Notification message" class="mt-1 text-sm text-gray-500">
                  {{ notificationStore.message }}
                </p>
              </div>
              <div class="ml-4 flex-shrink-0 flex">
                <button
                  type="button"
                  class="bg-white rounded-md inline-flex text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                  @click="close"
                >
                  <span class="sr-only">Close</span>
                  <XMarkIcon class="h-5 w-5" aria-hidden="true" />
                </button>
              </div>
            </div>
          </div>
        </div>
      </transition>
    </div>
  </div>
</template>
