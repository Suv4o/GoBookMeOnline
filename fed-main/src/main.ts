import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import { createPinia } from 'pinia'
import './assets/tailwind/index.css'
import { auth } from './config/firebase.config'
import { onAuthStateChanged } from 'firebase/auth'
import { useAuthStore } from './store/auth'
export const pinia = createPinia()
const appConfig = { isAppMounted: false }

const app = createApp(App)
app.provide('$auth', auth)
app.use(pinia)
app.use(router)

onAuthStateChanged(auth, user => {
  if (!user) {
    const useStoreAuth = useAuthStore(pinia)
    useStoreAuth.accessToken = ''
    useStoreAuth.accessTokenExpirationTime = 0
    useStoreAuth.user = null
  }

  if (!appConfig.isAppMounted) {
    appConfig.isAppMounted = true
    router.isReady().then(() => {
      app.mount('#app')
    })
  }
})
