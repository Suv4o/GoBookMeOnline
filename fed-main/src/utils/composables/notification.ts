import { useNotificationStore } from '../../store/notification'
import { NotificationState } from '../../store/notification'
let notificationTimeout = null as ReturnType<typeof setTimeout> | null

export function useNotification(
  notificationOptions: Pick<NotificationState, 'type' | 'title' | 'message'>,
  timeout = 8000
) {
  const notificationStore = useNotificationStore()
  const { type, title, message } = notificationOptions

  if (notificationTimeout) {
    clearTimeout(notificationTimeout)
  }

  notificationStore.isOpen = true
  notificationStore.type = type
  notificationStore.title = title
  notificationStore.message = message

  notificationTimeout = setTimeout(() => {
    notificationStore.isOpen = false
    notificationStore.type = null
    notificationStore.title = null
    notificationStore.message = null
  }, timeout)
}
