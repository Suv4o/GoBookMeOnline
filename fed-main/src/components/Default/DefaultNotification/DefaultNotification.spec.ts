import { render, cleanup } from '@testing-library/vue'
import { describe, it, expect } from 'vitest'
import DefaultNotification from './DefaultNotification.vue'
import { pinia } from '../../../main'
import { NotificationTypes, useNotificationStore } from '../../../store/notification'

describe('DefaultNotification', async () => {
  it('render component correctly', async () => {
    const useStoreNotification = useNotificationStore(pinia)

    useStoreNotification.isOpen = true
    useStoreNotification.type = NotificationTypes.Success
    useStoreNotification.title = 'Test Title'
    useStoreNotification.message = 'Test Message'

    const wrapper = render(DefaultNotification, {
      global: {
        plugins: [pinia],
      },
    })

    const notificationTitle = wrapper.getByTestId('Notification title')
    const notificationMessage = wrapper.getByTestId('Notification message')
    const notificationIcon = wrapper.getByTestId('Notification icon')

    expect(notificationTitle.textContent).toBe(useStoreNotification.title)
    expect(notificationMessage.textContent).toBe(useStoreNotification.message)
    expect(notificationIcon.childElementCount).toBe(1)

    cleanup()
  })

  it('snap shot matches', async () => {
    const wrapper = render(DefaultNotification, {
      global: {
        plugins: [pinia],
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
