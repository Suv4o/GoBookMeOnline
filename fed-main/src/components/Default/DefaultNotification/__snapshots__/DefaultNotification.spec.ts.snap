// Vitest Snapshot v1

exports[`DefaultNotification > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        aria-live="assertive"
        class="fixed inset-0 flex items-end px-4 py-6 pointer-events-none sm:p-6 sm:items-start z-30"
      >
        <div
          class="w-full flex flex-col items-center space-y-4 sm:items-end"
        >
          <transition-stub>
            <div
              class="max-w-sm w-full bg-white shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"
            >
              <div
                class="p-4"
              >
                <div
                  class="flex items-start"
                >
                  <div
                    class="flex-shrink-0"
                    data-testid="Notification icon"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-6 w-6 text-green-400"
                      fill="none"
                      stroke="currentColor"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                      />
                    </svg>
                    <!--v-if-->
                    <!--v-if-->
                    <!--v-if-->
                  </div>
                  <div
                    class="ml-3 w-0 flex-1 pt-0.5"
                  >
                    <p
                      class="text-sm font-medium text-gray-900"
                      data-testid="Notification title"
                    >
                      Test Title
                    </p>
                    <p
                      class="mt-1 text-sm text-gray-500"
                      data-testid="Notification message"
                    >
                      Test Message
                    </p>
                  </div>
                  <div
                    class="ml-4 flex-shrink-0 flex"
                  >
                    <button
                      class="bg-white rounded-md inline-flex text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                      type="button"
                    >
                      <span
                        class="sr-only"
                      >
                        Close
                      </span>
                      <svg
                        aria-hidden="true"
                        class="h-5 w-5"
                        fill="currentColor"
                        viewBox="0 0 20 20"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          clip-rule="evenodd"
                          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                          fill-rule="evenodd"
                        />
                      </svg>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-live="assertive"
      class="fixed inset-0 flex items-end px-4 py-6 pointer-events-none sm:p-6 sm:items-start z-30"
    >
      <div
        class="w-full flex flex-col items-center space-y-4 sm:items-end"
      >
        <transition-stub>
          <div
            class="max-w-sm w-full bg-white shadow-lg rounded-lg pointer-events-auto ring-1 ring-black ring-opacity-5 overflow-hidden"
          >
            <div
              class="p-4"
            >
              <div
                class="flex items-start"
              >
                <div
                  class="flex-shrink-0"
                  data-testid="Notification icon"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6 text-green-400"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                  <!--v-if-->
                  <!--v-if-->
                  <!--v-if-->
                </div>
                <div
                  class="ml-3 w-0 flex-1 pt-0.5"
                >
                  <p
                    class="text-sm font-medium text-gray-900"
                    data-testid="Notification title"
                  >
                    Test Title
                  </p>
                  <p
                    class="mt-1 text-sm text-gray-500"
                    data-testid="Notification message"
                  >
                    Test Message
                  </p>
                </div>
                <div
                  class="ml-4 flex-shrink-0 flex"
                >
                  <button
                    class="bg-white rounded-md inline-flex text-gray-400 hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-teal-500"
                    type="button"
                  >
                    <span
                      class="sr-only"
                    >
                      Close
                    </span>
                    <svg
                      aria-hidden="true"
                      class="h-5 w-5"
                      fill="currentColor"
                      viewBox="0 0 20 20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
